### PR TITLE
test: add card e2e tests

### PR DESCRIPTION
**Pull request by Seamoo13**
#### Please provide a brief summary of this pull request.
Add e2e tests for card component (core)

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by Seamoo13** [comment]
WIP until https://github.com/SAP/fundamental-ngx/pull/4143 merged

File: e2e/wdio/core/tests/card.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    browserIsIE,
+    getCSSPropertyByName,
+    waitForElDisplayed,
+    getElementArrayLength
+} from '../../driver/wdio';
+import { checkAttributeValueTrue, checkElArrIsClickable, checkElementText, checkElementTextValue } from '../../helper/assertion-helper';
+import CardData from '../fixtures/appData/card-content';
+
+describe('Card test suite:', function() {
+    const cardPg = new CardPo();
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/core/tests/card.e2e-spec.ts:15]: maybe leave it card or cardPage ?

File: e2e/wdio/helper/assertion-helper.ts [hunk 2]
```diff
@@ -99,3 +99,4 @@ export function checkElementTextValue(element: string, expectation): void {
 export function checkTextValueContain(oldValue: string, newValue: string): void {
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/helper/assertion-helper.ts:99]: I know it is not introduced by you but let's rename oldValue and newValue with string  and subString
Looks like these param names were inspired by specific test logic.

File: e2e/wdio/core/tests/card.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            checkAttributeValueTrue(cardPg.compactCardAttr, CardData.compactAttr);
+            checkElementText(cardPg.compactCardHeader);
+            checkElementTextValue(cardPg.compactCardListItems, CardData.cardListItemText);
+            checkElArrIsClickable(cardPg.compactCardHeader);
+            checkElArrIsClickable(cardPg.compactCardListItems);
+            if (!browserIsIE()) {
+                expect(getCSSPropertyByName(cardPg.compactCardListItems, CardData.fontSizeAttr).value)
+                    .toBe(CardData.compactFont);
+            }
+        });
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/core/tests/card.e2e-spec.ts:51]: skip check for IE comment?

File: e2e/wdio/core/pages/card.po.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    tableCardItemPrices = 'fd-card-table-example tbody td:nth-of-type(3)';
+    tableCardItemStatuses = 'fd-card-table-example td span';
+    // bar chart examples
+    barChartHeader = 'fd-bar-chart-list-card-example fd-card-header';
+    barChartTitle = 'fd-bar-chart-list-card-example fd-card-header h2';
+    barChartCounter = 'fd-bar-chart-list-card-example fd-card-header span';
+    barChartItems = 'fd-bar-chart-list-card-example li span';
+    barCharBars = 'fd-bar-chart-list-card-example fd-card-bar';
+
+    exampleAreaContainersArr = '.fd-doc-component';
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/core/pages/card.po.ts:52]: no need to add if the same as in the base  component

File: e2e/wdio/core/tests/card.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+        it('should check chart content', () => {
+            checkElementTextValue(cardPg.barChartItems, CardData.barChartItems);
+            expect(isElementDisplayed(cardPg.barCharBars)).toBe(true);
+        });
+    });
+
+    describe('Check orientation:', function() {
+        it('should check LTR and RTL', () => {
+            cardPg.checkRtlSwitch(cardPg.rtlSwitcherArr, cardPg.exampleAreaContainersArr);
```
Comment (AntonOlkhovskyi) [inline e2e/wdio/core/tests/card.e2e-spec.ts:130]: params can be removed

**Review by AntonOlkhovskyi — CHANGES_REQUESTED** [review]

File: e2e/wdio/core/tests/card.e2e-spec.ts (hunk 3 above)
Comment (Seamoo13) [reply e2e/wdio/core/tests/card.e2e-spec.ts:54]: added comment https://github.com/SAP/fundamental-ngx/issues/4310

**Review by AntonOlkhovskyi — APPROVED** [review]

File: e2e/wdio/core/tests/card.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        });
+
+        it('should check card content', () => {
+            checkElArrIsClickable(cardPage.cardListItems);
+            checkElementText(cardPage.cardListItems);
+            expect(getAttributeByName(cardPage.cardAttr, CardData.cardTypeAttr, 2)).toBe('list');
+        });
+    });
+
+    fdescribe('Compact examples:', function() {
```
Comment (nikvalor) [inline e2e/wdio/core/tests/card.e2e-spec.ts:40]: Can it be replaced with 'describe'?

Comment (valorkin) [reply e2e/wdio/core/tests/card.e2e-spec.ts:40]: has to be

**Review by nikvalor — APPROVED** [review]

File: e2e/wdio/helper/assertion-helper.ts [hunk 7]
```diff
@@ -96,6 +96,7 @@ export function checkElementTextValue(element: string, expectation): void {
     }
 }
 
-export function checkTextValueContain(oldValue: string, newValue: string): void {
-    expect(oldValue).toContain(newValue);
+export function checkTextValueContain(string: string, subString: string): void {
```
Comment (valorkin) [inline e2e/wdio/helper/assertion-helper.ts:99]: string -> str, please don't use standard keywords as variable names

**Review by valorkin — APPROVED** [review]

**Comment by Seamoo13** [comment]
reopened in https://github.com/SAP/fundamental-ngx/pull/4315 to fix commits

**Pull request closed without merging**